Update some packets

mdisprgm opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
Various changes. but the main one is for packets. 

## Motivation and Context
<!--- Why is this change/feature required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
1. Fix some components in`MinecraftPacketIds`. To follow the official name. - I'm not sure whether there is something I missed.
2. Implement **parts of** `UpdateTradePacket`. I believe it's not everything, but still, there are necessary things.

3. Use `Actor::startRiding` for `Actor.prototype.startRiding` instead of vftable offset. it seems it still works with Actor::startRiding even is a Mob.

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the code style of this project.
- [X] I have tested my changes and confirmed that they are working

karikera commented:
some classes override `startRiding`.
it can make issues.

`test.ts` checks the packet names and it can be failed if you didn't fix the test too.
did you check it passing?

mdisprgm commented:
> some classes override `startRiding`. it can make issues.
> 
> `test.ts` checks the packet names and it can be failed if you didn't fix the test too. did you check it passing?

OK. I reverted some changes. 

some packets' `getName` returns different name from its class name.
I reverted changes about that. 
but I'm still confused by `UpdateTradePacket`, and `UpdateEquipPacket`. their `getName`s return empty string. so, Can't I add the packets?

karikera commented:
You can just fix the test.
in fact, it would be better to add the test per new features.

mdisprgm commented:
oh.
I couldn't see the `wrongNames`. it is existing already.
so I added case of empty name.

karikera commented:
it seems good. Thanks.

personally, I hate some wasteful operations like duplicated comparing the string or using unnecessary function calls.
but it seems too meaningless to modify it for the just testing code.

mdisprgm commented:
OK.  thanks.
I will remember that to when edit other codes
